src/svm/interface/svm.c

```diff
@@ -2377,39 +2370,40 @@ PetscErrorCode SVMViewTrainingDataset(SVM svm,PetscViewer v)
 @*/
 PetscErrorCode SVMLoadTestDataset(SVM svm,PetscViewer v)
 {
-  MPI_Comm   comm;
+  MPI_Comm    comm;
 
-  Mat        Xt_test;
-  Mat        Xt_biased;
-  Vec        y_test;
+  Mat         X_test;
+  Mat         X_biased;
+  Vec         y_test;
 
-  PetscReal  bias;
-  PetscInt   mod;
+  PetscReal   bias;
+  PetscInt    mod;
 
-  const char *dataset_file;
   PetscBool  view_io,view_test_dataset;
 
   PetscFunctionBeginI;
   PetscValidHeaderSpecific(svm,SVM_CLASSID,1);
   PetscValidHeaderSpecific(v,PETSC_VIEWER_CLASSID,2);
 
   TRY( PetscObjectGetComm((PetscObject) svm,&comm) );
-  TRY( MatCreate(comm,&Xt_test) );
-  TRY( PetscObjectSetName((PetscObject) Xt_test,"Xt_test") );
+  TRY( MatCreate(comm,&X_test) );
+  /* Create matrix of test samples */
+  TRY( PetscObjectSetName((PetscObject) X_test,"X_test") );
+  TRY( PetscObjectSetName((PetscObject) X_test,"X_test") );
+  TRY( PetscObjectSetOptionsPrefix((PetscObject) X_test,"X_test_") );
+  TRY( MatSetFromOptions(X_test) );
+  /* Create label vector of test samples */
   TRY( VecCreate(comm,&y_test) );
   TRY( PetscObjectSetName((PetscObject) y_test,"y_test") );
 
-  TRY( SVMLoadDataset(svm,v,Xt_test,y_test) );
+  TRY( SVMLoadDataset(svm,v,X_test,y_test) );
```

Comment:
Again, it's not obvious why it is now X and not Xt.

Reply:
Mentioned above.